samples/snippets/src/main/java/com/example/spanner/admin/generated/CreateInstanceConfigSample.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner.admin.generated;
+
+// [START spanner_create_instance_config]
+import com.google.cloud.spanner.admin.instance.v1.InstanceAdminClient;
+import com.google.spanner.admin.instance.v1.CreateInstanceConfigRequest;
+import com.google.spanner.admin.instance.v1.InstanceConfig;
+import com.google.spanner.admin.instance.v1.ReplicaInfo;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+class CreateInstanceConfigSample {
+  static void createInstanceConfig() throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "projects/my-project";
+    String baseInstanceConfig = "nam11";
+    String instanceConfigId = "custom-instance-config4";
+    String instanceConfigName = "projects/my-project/instanceConfigs/custom-instance-config4";
+    createInstanceConfig(projectId, baseInstanceConfig, instanceConfigId, instanceConfigName);
+  }
+
+  static void createInstanceConfig(
+      String projectId, String baseInstanceConfig, String instanceConfigId, String instanceConfigName) throws IOException {
+    InstanceAdminClient instanceAdminClient = InstanceAdminClient.create();
```

Comment:
Not relevant in this PR, but I just thought of it when I saw this: We need additional samples for showing how to create the auto-generated clients when you want to use some non-standard options. The things that I can think of are:
1. Connecting to the emulator. Setting the `SPANNER_EMULATOR_HOST=localhost:9010` won't do it anymore. (Although we should consider creating some kind of standard helper method that customers can use that would do that, and add that method to the client library)
2. Connecting with custom credentials. That could then also be part of the above-mentioned helper method.

Reply:
Sure, I've taken a note. Will add these new samples in a new PR.